### ISSUE TITLE
Giveaway on BSO bot in non-native server can permanently delete items

### ISSUE THREAD

**Issue by KoDj2** [oldschoolgg/oldschoolbot#3264]
Item(s) used for a "=giveaway" in discord servers other than the primary OSB/BSO discord can cause the selected item(s) from the giveaway to be removed from the player bank, a shell of the giveaway will be performed and then an error will throw causing the item to be deleted and not returned to the user who was attempting to host the giveaway.
The bug may have been addressed as of today; it occurred from 12/7 through at least 12/9. As of this writing the function works on an outside server. Submitting bug report for evaluation and documentation purposes.
![giveawaybsobugfirerune](https://user-images.githubusercontent.com/93799326/145614008-f4904997-8e5e-4916-b6ab-75cb390c7c30.png)
![giveawaybsobug2](https://user-images.githubusercontent.com/93799326/145614303-4d30c494-49cc-49fb-8c45-d4f85f6860bb.png)

**Comment by themrrobert** [comment]
When the giveaway ends it should send you a PM saying noone entered the giveaway and refund your items, did you not get a pm or what happened when the auction ended? Are you sure the item was deleted and you werent refunded after it ended?

**Comment by KoDj2** [comment]
Currently I can't replicate the bug, it appears to be a transient issue and not permanent. Hopefully it self-resolved. Of note I was bug-testing as well so some of the parameters I inputted may have caused some issues but they don't appear to anymore so I'm not sure if there's a correlation or not. The same issue happened on different servers entirely upon further tests without attempting odd integers or equations.
I'm positive the bug removed all the items from the giveaway as it took all of the items and threw an error without a private message. I tested it with a stack of fire runes and all were deleted (see screenshot). I purchased 5, did the giveaway again, then purchased 5 more after the error and did another giveaway which only displayed the 5 I had purchased instead of 10 which I would've had if the items were returned to me.
I didn't lose anything substantial comparatively to what could've been lost, but it definitely deleted the items without returning them to my user bank.
![Screenshot_20211216-171824](https://user-images.githubusercontent.com/93799326/146463087-7072a192-2bc2-4dd2-a117-1d5224cfa904.png)
![Screenshot_20211216-170930](https://user-images.githubusercontent.com/93799326/146463096-e8ae690d-c05a-4fbe-9a58-0f53707a639a.png)
